connectivity/nanostack/sal-stack-nanostack/source/Security/TLS/tls_lib.c

```diff
@@ -2095,7 +2095,7 @@ void tls_master_key_cal(tls_heap_t *heap_ptr, sec_suite_t *tls_suite)
 {
     uint8_t *ptr;
     prf_sec_param_t *prf_ptr = shalib_prf_param_get();
-    uint8_t secret_buf[2 + 16 + 2 + 16];
+    static uint8_t secret_buf[2 + 16 + 2 + 16];
```

Comment:
Is `prf_ptr->secret` non local? 

I see lot of copying in  this tls_master_key_cal function but can't spot how anything that goes out of this function could point to this buffer (need more information).